shielder/contract/drink_tests/mod.rs

```diff
@@ -0,0 +1,226 @@
+mod utils;
+
+use drink::{
+    runtime::MinimalRuntime,
+    session::Session,
+    AccountId32,
+};
+
+use crate::{
+    contract::OpPub,
+    drink_tests::utils::{psp22::*, shielder::*},
+    mocked_zk::{ops::OpPriv, tests::merkle::MerkleTree},
+    types::Scalar,
+};
+
+#[drink::contract_bundle_provider]
+pub enum BundleProvider {}
+
+#[drink::test]
+fn deploy_single_deposit_single_withdraw() -> Result<(), Box<dyn std::error::Error>> {
+    const ALICE: AccountId32 = AccountId32::new([1u8; 32]);
+    const BOB: AccountId32 = AccountId32::new([2u8; 32]);
+
+    let mut session = Session::<MinimalRuntime>::new()?;
+    session.sandbox().mint_into(BOB, 1000000000000000).unwrap();
+
+    let shielder_address = deploy_shielder(&mut session)?;
+    let psp22_address = deploy_test_token(&mut session, 100)?;
+
+    let mut merkle_tree = MerkleTree::new();
+
+    // CREATE ACCOUNT
+    let user_shielded_data = create_shielder_account(
+        &mut session,
+        shielder_address.clone(),
+        psp22_address.clone(),
+        &mut merkle_tree,
+    )?;
+
+    // APPROVE TRANSFER
+    psp22_approve(
+        &mut session,
+        psp22_address.clone(),
+        shielder_address.clone(),
+        10,
+    )?;
+
+    // DEPOSIT
+    let user_shielded_data = shielder_update(
+        &mut session,
+        shielder_address.clone(),
+        UpdateOperation {
+            op_pub: OpPub::Deposit {
+                amount: 10,
+                token: Scalar {
+                    bytes: *(psp22_address.as_ref()),
+                },
```

Comment:
you might want to add some convenient constructor for `Scalar`

Reply:
Don't see if there is need for constructor, it is a struct holding bytes inside of itself. Possibly can provide From/Into trait impl, but don't want to bloat up Scalar